fdbclient/include/fdbclient/CommitTransaction.h

```diff
@@ -184,35 +255,53 @@ struct MutationRef {
 			StringRef empty;
 			if (!isEncrypted() && ar.protocolVersion().hasMutationChecksum() &&
 			    CLIENT_KNOBS->ENABLE_MUTATION_CHECKSUM) {
-				uint32_t c = populateChecksum();
-				StringRef cs = StringRef((uint8_t*)&c, 4);
-				uint8_t cType = this->typeWithChecksum();
-				serializer(ar, cType, param2, cs);
+				// Attach checksum at first, then attach acs index
+				this->populateChecksum();
+				uint8_t cType = this->setTypeWithChecksum(this->type);
+				uint32_t cs = this->checksum.get();
+				Standalone<StringRef> cEmpty = empty.withSuffix(StringRef((uint8_t*)&cs, 4));
+				if (CLIENT_KNOBS->ENABLE_ACCUMULATIVE_CHECKSUM && this->accumulativeChecksumIndex.present()) {
+					cType = this->setTypeWithAccumulativeChecksumIndex(cType);
```

Comment:
it seems fine to directly update `this->type` inside `setTypeWithAccumulativeChecksumIndex`.

Reply:
We may want to call setTypeWithAccumulativeChecksumIndex() outside the serialization, while we want to set the type only when the mutation is serialized.